src/main/resources/templates/db_scripts/publication-period/migratePublicationPeriod.sql

```diff
@@ -0,0 +1,132 @@
+CREATE OR REPLACE FUNCTION ${myuniversity}_${mymodule}.migrate_publication_period(
+    jsonb_data jsonb
+) RETURNS jsonb AS $$
+DECLARE
+    pub_period jsonb;
+    start_date text;
+    end_date text;
+    date_type text;
+    dates jsonb := '{}'::jsonb;
+BEGIN
+    pub_period := jsonb_data -> 'publicationPeriod';
+    start_date := pub_period ->> 'start';
+    end_date := pub_period ->> 'end';
+
+    -- Determine Date type
+    IF (start_date IS NOT NULL AND end_date IS NOT NULL) THEN
+        date_type := 'Multiple dates';
+    ELSIF (start_date IS NOT NULL OR end_date IS NOT NULL) THEN
+        date_type := 'Single known date/probable date';
+    ELSE
+        date_type := NULL;
+    END IF;
+
+    -- Build the JSONB Dates object
+    IF start_date IS NOT NULL THEN
+        dates := jsonb_set(dates, '{date1}', to_jsonb(start_date));
```

Comment:
Let's truncate date1 and date2 to 4 characters.

Reply:
done